### DISCUSSION TITLE
Algorithm for filtering the first two formants in Praat

### DISCUSSION THREAD

**Discussion by PHDAUTISM** [community/community#129706] [Programming Help]
### Body

I am a young researcher on autism and I am creating a novel methodology for analyzing speech patterns in individuals with autism spectrum disorder (ASD). This methodology involves using Praat to filter the first two formants from speech recordings. However, my coding skills are not strong, and I am struggling to find an algorithm that can automatically perform this task.

I would be grateful if anyone could point me in the direction of an algorithm that meets these requirements. If you have developed such an algorithm yourself, I would be happy to take a look at it.

I am particularly interested in algorithms that are efficient and easy to use. I would also be grateful for any advice on how to improve my coding skills.

Thank you for your time and consideration.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by floris-xlx** [comment]
load your audio file into an mp3 via any language you prefer,

make a function that can get the highest and lowest format level from the audio returning a tuple i.e

In rust for directional purposes
```rust
fn analyze_audio(audio_file: xxx) -> Result<u16, u16, anyhow::Error> {
    // take the audo thats loaded and extract all the audio levels generating a Vec<String>
    // take the Vec<String> and get the min and max value
    
    const audio_levels = {
       max_value: xx,
       min_value: xx
    }
}
```
then make a function to filter out into a by you desired treshold in between the range values the prior function gave you
then simply set a treshold and return only what exceeds the treshold and then do whatvever u want with it or save it,
